### PR TITLE
Added more examples of code-smells

### PR DESCRIPTION
**Pull request by PavlosIsaris**
Added examples for magic numbers, many parameter methods and unclear naming.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

File: client/src/pages/guide/english/agile/code-smells/index.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ It is important to understand that smelly code works, but is not of good quality
 #### Examples
 1. Duplicated code - Blocks of code that have been replicated across the code base. This may indicate that you need to generalize the code into a function and call it in two places, or it may be that the way the code works in one place is completely unrelated to the way it works in another place, despite having been copied.
 2. Large classes - Classes having too many lines of code. This may indicate that the class is trying to do too many things, and needs to be broken up into smaller classes.
+3. [Magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)) - Variables (or `if` statements) scattered in the code that hold numeric values with no apparent meaning. 
+4. Many-parameter methods - Methods that take more that 2 or 3 parameters usually indicate that the method does more than one thing and should be broken down to multiple meethods.
```
Comment (lynxlynxlynx) [inline client/src/pages/guide/english/agile/code-smells/index.md]: methods

File: client/src/pages/guide/english/agile/code-smells/index.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ It is important to understand that smelly code works, but is not of good quality
...
+5. Unclear method or variable naming - A programmer should be able to understand what a method does without looking at the method's body. If the name or the parameter naming is vague, it is an indicator that the method should be refacotred to reveal it's purpose in the code.
```
Comment (lynxlynxlynx) [inline client/src/pages/guide/english/agile/code-smells/index.md]: its

**Comment by lynxlynxlynx** [comment]
LGTM!

**Review by PavlosIsaris — COMMENTED** [review]
Looks good

**Review by HKuz — APPROVED** [review]

**Comment by HKuz** [comment]
LGTM! Thanks for adding that detail, @PavlosIsaris (and your QA @lynxlynxlynx) - the repo structure changed since you opened this, so I had to resolve the file conflicts on this end. 👃 🚮 ✨  Happy coding!

**Pull request merged**